### ISSUE TITLE
Impossible to install on Yunohost 3.3.4

### ISSUE THREAD

**Issue by bistule** [YunoHost-Apps/peertube_ynh#72]
Hi,

I've a problem when i try to install Peertube on yunohost, here is the log : 

```
args:
  app: peertube
  args: domain=try.me&email=peertube%try.me&is_public=1
  label: PeerTube
  no_remove_on_failure: false
ended_at: 2019-01-24 16:32:33.483005
env:
  YNH_APP_ARG_DOMAIN: tryt.me
  YNH_APP_ARG_EMAIL: peertube@try.me
  YNH_APP_ARG_IS_PUBLIC: 1
  YNH_APP_ID: peertube
  YNH_APP_INSTANCE_NAME: peertube
  YNH_APP_INSTANCE_NUMBER: '1'
  YNH_CWD: /var/cache/yunohost/from_file/peertube_ynh-7dc5b679ca17874a794742e27a2fd44ed3b904cf/scripts
  YNH_STDINFO: /tmp/tmpFubhje/stdinfo
error: Une erreur inattendue est survenue
operation: app_install
related_to:
- - app
  - peertube
started_at: 2019-01-24 16:31:59.831262
success: false

============

2019-01-24 16:32:04,765: WARNING - Installation of N - Node.js version management
2019-01-24 16:32:09,285: WARNING - 
                                                                           0,1%
                                                                           0,6%
#                                                                          1,6%
##                                                                         3,0%
#####                                                                      7,8%
#########                                                                 13,4%
#############                                                             19,4%
##################                                                        26,3%
######################                                                    31,2%
#########################                                                 36,1%
#############################                                             41,5%
###################################                                       49,5%
#######################################                                   54,9%
###########################################                               61,0%
##############################################                            64,2%
#################################################                         68,4%
######################################################                    75,7%
###########################################################               82,2%
###############################################################           88,6%
####################################################################      95,0%
######################################################################## 100,0%
2019-01-24 16:32:11,090: WARNING - Warning: apt-key output should not be parsed (stdout is not a terminal)
2019-01-24 16:32:33,349: WARNING - E: Sub-process /usr/bin/dpkg returned an error code (1)
2019-01-24 16:32:33,350: WARNING - Unable to install dependencies
2019-01-24 16:32:33,351: WARNING - !!
2019-01-24 16:32:33,351: WARNING -   peertube's script has encountered an error. Its execution was cancelled.
2019-01-24 16:32:33,352: WARNING - !!
2019-01-24 16:32:33,353: WARNING - 
2019-01-24 16:32:46,897: WARNING - E: Sub-process /usr/bin/dpkg returned an error code (1)
2019-01-24 16:32:47,802: WARNING - /usr/local/n wasn't deleted because it doesn't exist.
2019-01-24 16:32:47,803: WARNING - /etc/apt/sources.list.d/jessie-backports.list wasn't deleted because it doesn't exist.
2019-01-24 16:32:47,904: WARNING - dropdb : argument nom de la base de données requis mais manquant
2019-01-24 16:32:47,905: WARNING - Essayer « dropdb --help » pour plus d'informations.
2019-01-24 16:32:48,107: WARNING - dropuser : la suppression du rôle « peertube » a échoué : ERREUR:  le rôle « peertube » n'existe pas
2019-01-24 16:32:48,108: WARNING - Avoid deleting .
2019-01-24 16:32:48,109: WARNING - /etc/nginx/conf.d/try.me.d/peertube.conf wasn't deleted because it doesn't exist.
2019-01-24 16:32:48,812: WARNING - Remove the user peertube
2019-01-24 16:32:48,913: WARNING - userdel: cannot remove entry 'peertube' from /etc/passwd
```

Any idea ?

**Comment by quentinDupont** [comment]
On Yunohost 3.3.3 : 
```

Remove the user peertube

/etc/nginx/conf.d/videos.cloud117.fr.d/peertube.conf wasn't deleted because it doesn't exist.

/etc/apt/sources.list.d/jessie-backports.list wasn't deleted because it doesn't exist.

/usr/local/n wasn't deleted because it doesn't exist.

L’opération « Installer l’application « peertube » » a échouée ! Pour avoir de l’aide, merci de fournir le log complet de l’opération

!!

peertube's script has encountered an error. Its execution was cancelled.

!!

gyp ERR! not ok

gyp ERR! node-gyp -v v3.8.0

gyp ERR! node -v v8.15.0

gyp ERR! cwd /var/www/peertube/node_modules/rdf-canonize

gyp ERR! command "/opt/node_n/n/versions/node/8.15.0/bin/node" "/opt/node_n/n/versions/node/8.15.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"

gyp ERR! System Linux 4.9.144-xxxx-std-ipv6-64

gyp ERR! stack at Process.ChildProcess._handle.onexit (internal/child_process.js:198:12)

gyp ERR! stack at ChildProcess.emit (events.js:214:7)

gyp ERR! stack at emitTwo (events.js:126:13)

gyp ERR! stack at ChildProcess.onCpExit (/opt/node_n/n/versions/node/8.15.0/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:345:16)

gyp ERR! stack Error: `gyp` failed with exit code: 1

gyp ERR! configure error

gyp: Call to 'node -e "require('nan')"' returned exit status 0 while in binding.gyp. while trying to load binding.gyp

/bin/sh: BASH_XTRACEFD : 7 : valeur non valable pour un descripteur de fichier de trace

/bin/sh: BASH_XTRACEFD : 7 : valeur non valable pour un descripteur de fichier de trace

gyp info spawn args '-Goutput_dir=.' ]

gyp info spawn args 'build',

gyp info spawn args '--generator-output',

gyp info spawn args '--no-parallel',

gyp info spawn args '--depth=.',

gyp info spawn args '-Dnode_engine=v8',

gyp info spawn args '-Dmodule_root_dir=/var/www/peertube/node_modules/rdf-canonize',

gyp info spawn args '-Dnode_lib_file=/root/.node-gyp/8.15.0/<(target_arch)/node.lib',

gyp info spawn args '-Dnode_gyp_dir=/opt/node_n/n/versions/node/8.15.0/lib/node_modules/npm/node_modules/node-gyp',

gyp info spawn args '-Dnode_root_dir=/root/.node-gyp/8.15.0',

gyp info spawn args '-Dvisibility=default',

gyp info spawn args '-Dlibrary=shared_library',

gyp info spawn args '/root/.node-gyp/8.15.0/include/node/common.gypi',

gyp info spawn args '-I',

gyp info spawn args '/opt/node_n/n/versions/node/8.15.0/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',

gyp info spawn args '-I',

gyp info spawn args '/var/www/peertube/node_modules/rdf-canonize/build/config.gypi',

gyp info spawn args '-I',

gyp info spawn args 'make',

gyp info spawn args '-f',

gyp info spawn args 'binding.gyp',

gyp info spawn args [ '/opt/node_n/n/versions/node/8.15.0/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',

gyp info spawn /usr/bin/python2

gyp http 200 https://nodejs.org/download/release/v8.15.0/SHASUMS256.txt

gyp http GET https://nodejs.org/download/release/v8.15.0/SHASUMS256.txt

gyp http 200 https://nodejs.org/download/release/v8.15.0/node-v8.15.0-headers.tar.gz

gyp http GET https://nodejs.org/download/release/v8.15.0/node-v8.15.0-headers.tar.gz

gyp info using node@8.15.0 | linux | x64

gyp info using node-gyp@3.8.0

gyp info it worked if it ends with ok

/bin/sh: BASH_XTRACEFD : 7 : valeur non valable pour un descripteur de fichier de trace

sh: BASH_XTRACEFD : 7 : valeur non valable pour un descripteur de fichier de trace

/bin/sh: BASH_XTRACEFD : 7 : valeur non valable pour un descripteur de fichier de trace

Output:

Directory: /var/www/peertube/node_modules/rdf-canonize

Arguments:

Command: node-gyp rebuild

Exit code: 1

error /var/www/peertube/node_modules/rdf-canonize: Command failed.

warning Resolution field "@types/bluebird@3.5.21" is incompatible with requested version "@types/bluebird@3.5.18"

Executing: /lib/systemd/systemd-sysv-install enable postgresql

Synchronizing state of postgresql.service with SysV service script with /lib/systemd/systemd-sysv-install.

dpkg-preconfigure: impossible de réouvrir stdin :

debconf: Utilisation de l'interface Teletype en remplacement

debconf: (Cette interface a besoin d'un terminal de contrôle.)

debconf: Impossible d'initialiser l'interface : Readline

debconf: Utilisation de l'interface Readline en remplacement

debconf: (L'interface dialog ne fonctionnera pas avec un terminal rustique (« dumb »), un tampon shell d'Emacs ou sans terminal de contrôle.)

debconf: Impossible d'initialiser l'interface : Dialog

Warning: apt-key output should not be parsed (stdout is not a terminal)

0,2% 0,6% # 2,6% ### 5,2% ######## 11,8% ############ 18,0% ################# 24,8% ###################### 31,6% ########################### 38,4% ################################ 45,3% #################################### 50,8% ######################################### 57,3% ############################################# 63,5% ################################################## 69,5% ######################################################## 78,9% ############################################################# 85,4% ################################################################## 91,8% ###################################################################### 98,2% ######################################################################## 100,0%

Installation of N - Node.js version management
```

**Comment by thepra** [comment]
I have issue also, installing it on my Raspberry Pi:
2019-01-29 21:03:41,982: WARNING - Installation of N - Node.js version management
2019-01-29 21:03:47,713: WARNING -
0.1%
0.2%
0.6%
1.3%
2.7%
5.6%

####### 10.2%
############# 18.9%
################## 25.6%
####################### 32.5%
################################# 46.0%
###################################### 52.9%
########################################## 59.7%
############################################### 66.4%
#################################################### 73.2%
######################################################### 80.0%
############################################################## 86.8%
################################################################### 93.6%
######################################################################## 100.0%
2019-01-29 21:03:58,852: WARNING - Warning: apt-key output should not be parsed (stdout is not a terminal)
2019-01-29 21:04:33,870: WARNING - debconf: unable to initialize frontend: Dialog
2019-01-29 21:04:33,873: WARNING - debconf: (Dialog frontend will not work on a dumb terminal, an emacs shell buffer, or without a controlling terminal.)
2019-01-29 21:04:33,875: WARNING - debconf: falling back to frontend: Readline
2019-01-29 21:04:33,878: WARNING - debconf: unable to initialize frontend: Readline
2019-01-29 21:04:33,880: WARNING - debconf: (This frontend requires a controlling tty.)
2019-01-29 21:04:33,883: WARNING - debconf: falling back to frontend: Teletype
2019-01-29 21:04:33,985: WARNING - dpkg-preconfigure: unable to re-open stdin:
2019-01-29 21:04:41,116: WARNING - Synchronizing state of postgresql.service with SysV service script with /lib/systemd/systemd-sysv-install.
2019-01-29 21:04:41,123: WARNING - Executing: /lib/systemd/systemd-sysv-install enable postgresql
2019-01-29 21:05:07,518: WARNING - Usage: yarn [options]
2019-01-29 21:05:07,522: WARNING -
2019-01-29 21:05:07,524: WARNING - yarn: error: no such option: --production
2019-01-29 21:05:07,627: WARNING - !!
2019-01-29 21:05:07,629: WARNING - peertube's script has encountered an error. Its execution was cancelled.
2019-01-29 21:05:07,630: WARNING - !!
2019-01-29 21:05:07,632: WARNING -

**Comment by anmol26s** [comment]
Do
`apt install apt-transport-https`

**Comment by isAAAc** [comment]
apt-transport-https is already the newest version (1.4.9)

**Comment by anmol26s** [comment]
I have updated the Peertube. Can you(everyone) install/upgrade again and see if it works?

**Comment by isAAAc** [comment]
install fails : https://paste.yunohost.org/raw/atinejikok
:/

**Comment by isAAAc** [comment]
i think i have a problem with my ca-certificates on my server krashboyz.org, i have a "similar" trouble on the install of roundcube https://github.com/YunoHost-Apps/roundcube_ynh/issues/49  ... 

i try the peertube instal on an other ynh server to be sure

**Comment by isAAAc** [comment]
ok, back : install without any trouble on media.emancipahost.nohost.me , so i thnik i have a ca-certificates trouble on my yunohost install on krashboyz.org , i'll open a new issue on main yunohost and refer it here. Consider this issue as pending for now.
thx for your work @anmol26s ;)

**Comment by isAAAc** [comment]
topic opened on the forum : https://forum.yunohost.org/t/probleme-avec-les-ca-certificates/7058

**Comment by isAAAc** [comment]
ok, everything is ok now, install of peertube is OK
i had a bad version of librtmp1 (since an experimental action in parallel of ynh install :/ )
```apt install librtmp1=2.4+20151223.gitfa8646d.1-1+b1```  and everything gone good, 
special thx to @alexAubin and the #yunohost-dev irc chan for the support <3

i don't close this issue because i didn't opened it :/

**Comment by wioxjk** [comment]
It does not work if you installed the Yuno with the .iso, but if you install it on top of a debian installation - it works.

**Comment by yalh76** [comment]
Feel free to reopen this issue if needed
